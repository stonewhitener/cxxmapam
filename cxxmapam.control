comment = 'table access method using C++ std::map'
default_version = '1.0'
module_pathname = '$libdir/cxxmapam'
relocatable = true
