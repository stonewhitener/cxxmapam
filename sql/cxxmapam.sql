CREATE EXTENSION cxxmapam;
CREATE TABLE cxxmap_tab (foo int, bar "char") USING cxxmap;
SELECT * FROM cxxmap_tab;
INSERT INTO cxxmap_tab VALUES (123, 'A');
SELECT * FROM cxxmap_tab;
INSERT INTO cxxmap_tab VALUES (456, 'B');
SELECT * FROM cxxmap_tab;

-- Clean up
DROP TABLE cxxmap_tab;
