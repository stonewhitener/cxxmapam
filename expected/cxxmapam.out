CREATE EXTENSION cxxmapam;
CREATE TABLE cxxmap_tab (foo int, bar "char") USING cxxmap;
SELECT * FROM cxxmap_tab;
 foo | bar 
-----+-----
(0 rows)

INSERT INTO cxxmap_tab VALUES (123, 'A');
SELECT * FROM cxxmap_tab;
 foo | bar 
-----+-----
 123 | A
(1 row)

INSERT INTO cxxmap_tab VALUES (456, 'B');
SELECT * FROM cxxmap_tab;
 foo | bar 
-----+-----
 123 | A
 456 | B
(2 rows)

-- Clean up
DROP TABLE cxxmap_tab;
